bears/python/MypyBear.py

```diff
@@ -0,0 +1,155 @@
+        inverse=True),
+    'allow_untyped_calls': FlagInfo(
+        arg='--disallow-untyped-calls',
+        doc='Allow calling functions without type annotations from '
+            'typed functions.',
+        inverse=True),
+    'skip_untyped_functions': FlagInfo(
+        arg='--check-untyped-defs',
+        doc='Do not check the interior of functions without type annotations.',
+        inverse=True),
+    'strict_optional': FlagInfo(
+        arg='--strict-optional',
+        doc='Enable experimental strict checks related to Optional types. See '
+            '<http://mypy-lang.blogspot.com.es/2016/07/mypy-043-released.html>'
+            ' for an explanation.',
+        inverse=False),
+}
+
+
+def add_param_docs(param_map):
+    """Append documentation from FLAG_MAP to a function's docstring.
+
+    :param param_map:
+        A mapping of argument names (strings) to FlagInfo objects.
+    :return:
+        A decorator that appends flag information to a function's docstring.
+    """
+    def decorator(func):
+        func.__doc__ = textwrap.dedent(func.__doc__) + '\n'.join(
+            ':param {}:\n{}'.format(name, textwrap.indent(arg.doc, '    '))
+            for name, arg in param_map.items())
+        return func
+    return decorator
+
+
+@linter(executable=sys.executable,
+        prerequisite_check_command=(sys.executable, '-m', 'mypy', '-V'),
+        )
+class MypyBear:
+    """
+    Type-checks your Python files!
+
+    Checks optional static typing using the mypy tool.
+    See <http://mypy.readthedocs.io/en/latest/basics.html> for info on how to
+    add static typing.
+    """
+
+    LANGUAGES = {"Python", "Python 2", "Python 3"}
+    AUTHORS = {'Petr Viktorin'}
+    REQUIREMENTS = {PipRequirement('mypy', '0.*')}
+    AUTHORS_EMAILS = {'encukou@gmail.com'}
+    LICENSE = 'AGPL-3.0'
+
+    # This detects typing errors, which is pretty unique -- it doesn't
+    # make sense to add a category for it.
+    CAN_DETECT = set()
+
+    @add_param_docs(FLAG_MAP)
+    def create_arguments(self, filename, file, config_file,
+                         language: str="Python 3",
+                         python_version: str=None,
+                         allow_untyped_functions: bool=True,
+                         allow_untyped_calls: bool=True,
+                         skip_untyped_functions: bool=True,
+                         strict_optional: bool=False,
+                         ):
+        """
+        :param language:
+            Set to "Python" or "Python 3" to check Python 3.x source.
+            Use "Python 2" for Python 2.x.
+        :param python_version:
+            Set the specific Python version, e.g. "3.5".
+        """
+        args = ['-m', 'mypy']
+        if language in ('Python 3', 'Python'):
+            pass
+        elif language == 'Python 2':
+            args.append('--py2')
+        else:
+            self.err(
+                'Language needs to be "Python", "Python 2" or "Python 3"')
```

Comment:
Not really intentionally – it's because I don't know of a good way to signal an error from `create_arguments`.
I'll treat that as a feature and add the note you're suggesting.